### PR TITLE
refactor (webapi): exchange ActorSelection with ActorRef

### LINKED ISSUE #1167: Refactor ActorSelection to ActorRef
In a previous PR, I changed `ActorSelection` to `ActorRef` in the `Responders`. There are a few instances of `ActorSelection` still present, which should also be refactored to `ActorRef`.

Reasoning:

For a message to be sent to an actor, a reference (`ActorRef`) is necessary. With `ActorSelection` messages can be sent to actors without this reference, as the ActorSystem is taking care of finding the reference. Getting rid of `ActorSelection` simplifies things a bit, as it removes one step that can fail. Also, the ActorSystem has less work to perform.

### PR DESCRIPTION
**Pull request by subotic**
### Summary
- [x] exchange ActorSelection with ActorRef
- [x] refactor all routes
- [x] refactor all responders
- [x] update docs

### Issues
- resolves #1167

### CODE REVIEW

**Comment by subotic** [comment]
A lot of files changed, but always the same things.

File: webapi/src/main/scala/org/knora/webapi/responders/Responder.scala [hunk 1]
```diff
@@ -48,15 +48,25 @@ object Responder {
...
 
+/**
+  * Data needed to be passed to each responder.
+  *
+  * @param system the actor system.
+  * @param applicationStateActor the application state actor ActorRef.
+  * @param responderManager the responder manager ActorRef.
+  * @param storeManager the store manager ActorRef.
+  */
+case class ResponderData(system: ActorSystem, applicationStateActor: ActorRef, responderManager: ActorRef, storeManager: ActorRef)
```
Comment (benjamingeer) [inline webapi/src/main/scala/org/knora/webapi/responders/Responder.scala]: I like this.

File: webapi/src/main/scala/org/knora/webapi/routing/KnoraRoute.scala [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+  * Data needed to be passed to each route.
+  *
+  * @param system the actor system.
+  * @param applicationStateActor the application state actor ActorRef.
+  * @param responderManager the responder manager ActorRef.
+  * @param storeManager the store manager ActorRef.
+  */
+case class KnoraRouteData(system: ActorSystem, applicationStateActor: ActorRef, responderManager: ActorRef, storeManager: ActorRef)
```
Comment (benjamingeer) [inline webapi/src/main/scala/org/knora/webapi/routing/KnoraRoute.scala]: I like this, too.

File: webapi/src/main/scala/org/knora/webapi/routing/KnoraRoute.scala [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    implicit protected val timeout: Timeout = settings.defaultTimeout
+    implicit protected val executionContext: ExecutionContext = system.dispatchers.lookup(KnoraDispatchers.KnoraActorDispatcher)
+    implicit protected val stringFormatter: StringFormatter = StringFormatter.getGeneralInstance
+
+    protected val applicationStateActor: ActorRef = routeData.applicationStateActor
+    protected val storeManager: ActorRef = routeData.storeManager
+    protected val log = akka.event.Logging(system, this.getClass)
+
+    /* define required method */
+    def knoraApiPath: Route
```
Comment (benjamingeer) [inline webapi/src/main/scala/org/knora/webapi/routing/KnoraRoute.scala]: Please add a ScalaDoc comment for this abstract method.

File: webapi/src/main/scala/org/knora/webapi/responders/Responder.scala (hunk 1 above)
Comment (subotic) [reply webapi/src/main/scala/org/knora/webapi/responders/Responder.scala]: should make refactoring in the future much easier :-)

File: webapi/src/main/scala/org/knora/webapi/routing/KnoraRoute.scala (hunk 3 above)
Comment (subotic) [reply webapi/src/main/scala/org/knora/webapi/routing/KnoraRoute.scala]: done

**Comment by subotic** [comment]
Thanks for the review :-)

**Pull request merged**